pkg/sql/sqlbase/rowwriter.go

```diff
@@ -842,24 +842,36 @@ func (ru *RowUpdater) UpdateRow(
 	}
 
 	// Update secondary indexes.
-	for i, newSecondaryIndexEntry := range newSecondaryIndexEntries {
-		secondaryIndexEntry := secondaryIndexEntries[i]
+	// We're iterating through all of the indexes, which should have corresponding entries in both oldSecondaryIndexEntries
+	// and newSecondaryIndexEntries. Inverted indexes could potentially have more entries at the end of both and we will
+	// update those separately.
+	for i, index := range ru.Helper.Indexes {
+		oldSecondaryIndexEntry := oldSecondaryIndexEntries[i]
+		newSecondaryIndexEntry := newSecondaryIndexEntries[i]
+
```

Comment:
can you not change the code below and instead use

if index.Type == IndexDescriptor_INVERTED {
.....
continue
}

that way the inverted case is separated out and is easier to understand and you can add a comment for why you're deleting the old and appending the new value in this special code. Thanks!